proposed/phpdoc.md

```diff
@@ -0,0 +1,2215 @@
+or Properties. Such a compound statement can only have one DocBlock while several
+items are represented.
+
+#### Examples
+
+```php
+/** @var int $int This is a counter. */
+$int = 0;
+
+// there should be no docblock here
+$int++;
+```
+
+Or:
+
+```php
+class Foo
+{
+  /** @var string|null Should contain a description */
+  protected $description = null;
+
+  public function setDescription($description)
+  {
+      // there should be no docblock here
+      $this->description = $description;
+  }
+}
+```
+
+Another example is to document the variable in a foreach explicitly; many IDEs
+use this information to help you with auto-completion:
+
+```php
+/** @var \Sqlite3 $sqlite */
+foreach($connections as $sqlite) {
+    // there should be no docblock here
+    $sqlite->open('/my/database/path');
+    <...>
+}
+```
+
+Even compound statements may be documented:
+
+```php
+class Foo
+{
+  protected 
+      /**
+       * @var string Should contain a description
+       */
+      $name, 
+      /**
+       * @var string Should contain a description
+       */
+      $description;
+
+}
+```
+
+Or constants:
+
+```php
+class Foo
+{
+  const 
+      /**
+       * @var string Should contain a description
+       */
+      MY_CONST1 = "1", 
+      /**
+       * @var string Should contain a description
+       */
+      MY_CONST2 = "2";
+
+}
+```
+
+### 8.25. @version
+
+The @version tag is used to denote some description of "versioning" to an
+element.
+
+#### Syntax
+
+    @version ["Semantic Version"] [<description>]
+
+#### Description
+
+Documents the current "version" of any element.
+
+This information can be used to generate a set of API Documentation where the
+consumer is informed about elements at a particular version.
+
+It is RECOMMENDED that the version number matches a semantic version number as
+described in the [Semantic Versioning Standard version 2.0][SEMVER2].
+
+Version vectors from Version Control Systems are also supported, though they
+MUST follow the form:
+
+    name-of-vcs: $vector$
+
+A description MAY be provided, for the purpose of communicating any additional
+version-specific information.
+
+The @version tag MAY NOT be used to show the last modified or introduction
+version of an element, the @since tag SHOULD be used for that purpose.
+
+#### Examples
+
+```php
+/**
+ * File for class Foo
+ * @version 2.1.7 MyApp
+ *          (this string denotes the application's overall version number)
+ * @version @package_version@
+ *          (this PEAR replacement keyword expands upon package installation)
+ * @version $Id$
+ *          (this CVS keyword expands to show the CVS file revision number)
+ */
+
+/**
+ * This is Foo
+ */
+class Foo
+{
+  <...>
+}
+```
+
+## Appendix A. Types
+
+### ABNF
+
+A Type has the following [ABNF][RFC5234] definition:
+
+    type-expression  = type *("|" type)
+    type             = class-name / keyword / array
+    array            = (type / array-expression) "[]" / generic
```

Comment:
As `type` can be an array, this rule is recursive: e. g. `SomeClass[][]`.
Is this intended? If yes, it should be described in the "Arrays" section below.